### PR TITLE
feat(sumo): LaneChangeMode.OFF should do no lane changes at all

### PR DESCRIPTION
**Pull request by kschrab**
## Description

* Previously, `LaneChangeMode.OFF` still did strategic lane changes, e.g. to follow a route.
* Now, `LaneChangeMode.OFF` disables all lane changes in SUMO, an additional `LaneChangeMode.FOLLOW_ROUTE` reprises the previous behavior, with all lane change reasons turned off, except strategic lane changes for route-following.
* Furthermore, disallow negative numbers when calculating the target lane for a lane change.

Offtopic:
* RTI real time synchronization (`-b`) now works much better when pausing the simulation. Then it will not try to catch up to meet the configured real-time requirement. Instead, it will only do real-time sync for the last 1 second.

What is this PR about?

 
## Affected parts of the online documentation

Changes in the documentation required?

Not really, but https://eclipse.dev/mosaic/docs/develop_applications/operating_system/ is apparently broken (there is still milliseconds passed to `changeLane(..)` instead of nanoseconds.

## Definition of Done

**Prerequisites**

- [x] You have read CONTRIBUTING.md carefully.
- [x] You have signed the [Contributor License Agreement](http://www.eclipse.org/legal/CLA.php).
- [x] Your GitHub user id is linked with your Eclipse Account.

**Required**

- [x] The title of this merge request follows the scheme `type(scope): description`  (in the style of [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary))
- [x] You have assigned a suitable label to this pull request (e.g., `enhancement`, or `bugfix`)
- [ ] `origin/main` has been merged into your Fork.
- [ ] Coding guidelines have been followed (see CONTRIBUTING.md).
- [ ] All checks on GitHub pass.
- [ ] All tests on [Jenkins](https://ci.eclipse.org/mosaic/job/mosaic/) pass.

**Requested** (can be enforced by maintainers)

- [ ] New functionality is covered by unit tests or integration tests. Code coverage must not decrease.
- [ ] If a bug has been fixed, a new unit test has been written (beforehand) to prove misbehavior
- [ ] There are no new SpotBugs warnings.

### CODE REVIEW

**Review by schwepmo — APPROVED** [review]
LGTM!

File: fed/mosaic-sumo/src/main/java/org/eclipse/mosaic/fed/sumo/bridge/api/complex/SumoLaneChangeMode.java [hunk 1]
```diff
@@ -51,16 +55,10 @@ public static SumoLaneChangeMode translateFromEnum(final LaneChangeMode laneChan
                 mode.setStrategicChanges(true, true);
                 mode.setCooperativeChanges(true, false);
                 mode.setSpeedChanges(true, false);
-                mode.setRightDriveChanges(true, true);
+                mode.setRightDriveChanges(true, false);
                 mode.setRespectOtherDrivers(RESPECT_SPEED_GAPS_OF_OTHER_DRIVERS_DO_NOT_ADAPT_SPEED);
                 break;
             case PASSIVE:
```
Comment (schwepmo) [inline fed/mosaic-sumo/src/main/java/org/eclipse/mosaic/fed/sumo/bridge/api/complex/SumoLaneChangeMode.java:61]: please check passive and cooperative

Comment (kschrab) [reply fed/mosaic-sumo/src/main/java/org/eclipse/mosaic/fed/sumo/bridge/api/complex/SumoLaneChangeMode.java:61]: The only difference between PASSIVE and COOPERATIVE was that the second bool in `setRightDriveChanges()` was set to `true` for PASSIVE. If the second boolean is set to true, than lane-changes via TraCI(MOSAIC API) are ignored if the lane-change model itself wants to keep the right lane. Ignoring TraCI at this point is not helpful imo. Only strategic lane changes should override lane-changes via TraCI, to keep route-following alive. 

Tl;dr from my pov it's fine as it is here now.
